### PR TITLE
Support for container transition

### PR DESCRIPTION
**Pull request by tbaranes**
Related to #188 

This PR contains only one new class: `Container` (need to find a better name) allowing the usage of `IBAnimatable` animators to make transition between childViewController. The usage is really straightforward:

``` swift
let transitionContext = ContainerTransition(animationType: .Fold(direction: direction, params: []),
                                                    container: containerView,
                                                    parentViewController: self,
                                                    fromViewController: fromViewController,
                                                    toViewController: toViewController) {
      self.currentViewController = toViewController
}
transitionContext.animate()
```

More details:
- Doesn't support interactive gesture
- Doesn't support for interface builder
- If nil is passed as `fromViewController`, it will set the `toViewController` as default without animation
- Handle all the logic to transit from a view controller to a new one, with an animation
- Support every `TransitionAnimationType`

That feature doesn't really follow the main goal of `IBAnimatable` (a library for designers), but I'm sure it's worth it for every developers using it programatically.  What do you think?

If you are ok to merge it, I will update the CHANGELOG, README, and also add an example in the demo project.

### CODE REVIEW

File: IBAnimatable/Container.swift [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    
+    parentViewController?.view.userInteractionEnabled = false
+    let animator = AnimatorFactory.generateAnimator(unwrappedAnimationType)
+    animator.animateTransition(self)
+  }
+}
+
+// MARK: - UIViewControllerContextTransitioning
+
+extension ContainerTransition {
```
Comment (JakeLin) [inline IBAnimatable/Container.swift]: Can we move `UIViewControllerContextTransitioning` in this `extension`?

File: IBAnimatable/Container.swift [hunk 2]
```diff
@@ -0,0 +1,110 @@
+//
+//  Created by Tom Baranes on 06/05/16.
+//  Copyright © 2016 Jake Lin. All rights reserved.
+//
+
+import UIKit
+
+class ContainerTransition: NSObject, UIViewControllerContextTransitioning {
```
Comment (JakeLin) [inline IBAnimatable/Container.swift]: Does it need to be `public`?

**Comment by JakeLin** [comment]
@tbaranes thanks for your PR, I don't have better idea to call it, may call `ContainerControllerAnimatedTransition` or just `ContainerTransition`? it is your call to name it. 

IB doesn't support ContainerView either, and we have to use code to support / switch multiple viewControllers, I actually like this PR because it can enable all `***Animator`s to use in sub controllers, not only for Push/Pop and Present/Dismiss transitions. That one principle when we architectured our Transition Animators / Controllers, they are all standard transition controllers conforming to `UIViewControllerAnimatedTransitioning`, we can use them in Apple's standard way for transitions.

**Comment by tbaranes** [comment]
Happy to see you like it! For the name, I was thinking about `ContainerTransition` too, so it will probably be the final name. 

l will add an example, and update the docs this weekend 🎉

**Comment by JakeLin** [comment]
cool, thanks @tbaranes  🎉

**Comment by tbaranes** [comment]
Update of this PR:
- Add a `UITabBar` using `ContainerTransition` example, it's accessible from the transitions tableView. Each `UITabBarItem` will be show with different animation: Explode, Fold, Portal, and NatGeo
- Update the CHANGELOG
- Didn't update the README since we are going to change its architecture

Ready to go 🎉

**Comment by JakeLin** [comment]
Cool, please merge it. 👍

**Pull request merged**